relayer/ibc-client.go

```diff
@@ -91,13 +94,9 @@ func (c *Chain) InjectTrustedFields(dst *Chain, header *tmclient.Header) (*tmcli
 	// make copy of header stored in mop
 	h := *(header)
 
-	dstH, err := dst.GetLatestLightHeight()
```

Comment:
this logic didn't make any sense. We should just query using the latest state possible